core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2762,7 +2763,12 @@ class ReplicaManager(val config: KafkaConfig,
         replicaFetcherManager.shutdownIdleFetcherThreads()
         replicaAlterLogDirsManager.shutdownIdleFetcherThreads()
 
-        remoteLogManager.foreach(rlm => rlm.onLeadershipChange(leaderChangedPartitions.asJava, followerChangedPartitions.asJava, localChanges.topicIds()))
+        val isTieredStateEnabled = getLogConfig(localChanges.deletes.asScala.head).get.getBoolean(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG)
+
+        remoteLogManager.foreach(rlm => {
+          rlm.validateCopyAndExpireTasks(isTieredStateEnabled, leaderChangedPartitions.asJava, localChanges.topicIds())
+          rlm.onLeadershipChange(leaderChangedPartitions.asJava, followerChangedPartitions.asJava, localChanges.topicIds())
```

Comment:
This method delegates to two further methods in RLM called doHandleLeaderPartition and doHandleFollowerPartition. Is it not better to try and put the logic of validateCopyAndExpireTasks in them?

Reply:
Indeed, its better that way, moved it into that method.